internal/services/network/network_manager_scope_connection_resource.go

```diff
@@ -0,0 +1,267 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/network/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+	"github.com/tombuildsstuff/kermit/sdk/network/2022-05-01/network"
+)
+
+type ManagerScopeConnectionModel struct {
+	Name             string `tfschema:"name"`
+	NetworkManagerId string `tfschema:"network_manager_id"`
+	ConnectionState  string `tfschema:"connection_state"`
+	Description      string `tfschema:"description"`
+	ResourceId       string `tfschema:"resource_id"`
+	TenantId         string `tfschema:"tenant_id"`
+}
+
+type ManagerScopeConnectionResource struct{}
+
+var _ sdk.ResourceWithUpdate = ManagerScopeConnectionResource{}
+
+func (r ManagerScopeConnectionResource) ResourceType() string {
+	return "azurerm_network_manager_scope_connection"
+}
+
+func (r ManagerScopeConnectionResource) ModelObject() interface{} {
+	return &ManagerScopeConnectionModel{}
+}
+
+func (r ManagerScopeConnectionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return validate.NetworkManagerScopeConnectionID
+}
+
+func (r ManagerScopeConnectionResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"network_manager_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NetworkManagerID,
+		},
+
+		"resource_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
```

Comment:
we don't expose `resource_id` - what is this, a scope?

Reply:
modifies it to `target_scope_id`, it should be either Subscription ID or Management Group ID.